homeassistant/core.py

```diff
@@ -352,6 +352,9 @@ def async_add_job(
         if target is None:
             raise ValueError("Don't call async_add_job with None")
 
+        if asyncio.iscoroutine(target):
+            return self.async_create_task(cast(Coroutine, target))
```

Comment:
I'm not sure if this was the intent, but the task won't be tracked because we don't add it to 

```
        if self._track_task:
            self._pending_tasks.append(task)
```

Reply:
It will be tracked if we're tracking tasks. Which is how the previous behavior was too (as i recall, on my phone)

Reply:
We call hass.async_create_task which also tracks tasks.